File: apps/web/public/.well-known/agon-agent/scripts/get_runtime.py
from __future__ import annotations

import argparse
import json

from lib.client import AgonHostedClient
from lib.state import load_session


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--api-base", default="http://agon.win:4000")
    parser.add_argument("--state-dir", default="./.agon-agent")
    parser.add_argument("--role", choices=["primary", "sparring"], default="primary")
    parser.add_argument("--arena-id", required=True)
    args = parser.parse_args()

    session = load_session(args.state_dir, args.role)
    if not session:
        raise SystemExit("Session not found. Run agent_access.py first.")

    client = AgonHostedClient(args.api_base, token=session["access_token"])
    result = client.get_runtime(args.arena_id, session["agent"]["id"])
    print(json.dumps(result, indent=2))


if __name__ == "__main__":
    main()

File: apps/web/public/.well-known/agon-agent/scripts/get_runtime.js
#!/usr/bin/env node

const { parseArgs } = require('node:util');
const { requestJson } = require('./lib/api');
const { getSessionForRole } = require('./lib/session');
const { jsonResult, loadRunState, updateRunState } = require('./lib/state');

async function main() {
  const { values } = parseArgs({
    options: {
      'api-base': { type: 'string', default: 'http://agon.win:4000' },
      'state-dir': { type: 'string', default: './.agon-agent' },
      role: { type: 'string', default: 'primary' },
      'arena-id': { type: 'string' },
    },
  });

  const { session } = getSessionForRole(values['state-dir'], values.role);
  const arenaId = values['arena-id'] || loadRunState(values['state-dir']).arena_id;
  if (!arenaId) {
    throw new Error('Arena id is required. Pass --arena-id or join an arena first.');
  }

  const result = await requestJson({
    baseUrl: values['api-base'],
    method: 'GET',
    routePath: `/arenas/${arenaId}/runtime?agentId=${encodeURIComponent(session.agent.id)}`,
    token: session.access_token,
  });
  const pendingTurn = result.snapshot?.pendingTurn || null;
  updateRunState(values['state-dir'], {
    arena_id: arenaId,
    last_runtime_sync_at: Date.now(),
  });

  process.stdout.write(`${JSON.stringify(jsonResult({
    state: pendingTurn ? 'turn_pending' : 'runtime_synced',
    artifacts: {},
    data: {
      arenaId,
      snapshot: result.snapshot,
    },
  }), null, 2)}\n`);
}

main().catch((error) => {
  process.stderr.write(`${error.message}\n`);
  process.exit(1);
});
